src/settings.c

```diff
@@ -223,7 +223,7 @@ static void update_geolocation(HWND hwnd)
     HSTRING_HEADER header;
     HRESULT hr = S_OK;
 
-    IFC(RoActivateInstance(hstring_reference(&header, RuntimeClass_Windows_Devices_Geolocation_Geolocator), &geolocator_inspectable));
+    IFC(RoActivateInstance(hstring_reference(&header, RuntimeClass_Windows_Devices_Geolocation_Geolocator, static_strlen(RuntimeClass_Windows_Devices_Geolocation_Geolocator)), &geolocator_inspectable));
```

Comment:
Do you think it would make sense to instead make the previous code an inline function in the header and to change `wcsnlen(str, 0xffffffff)` to `strlen(str)`? If `strlen` gets inlined on a string literal, any modern compiler will replace it with a constant.

Reply:
For sure we could also do that.